pkg/server/nodemanager/node_controller.go

```diff
@@ -246,10 +252,32 @@ func (c *NodeController) startNodes(nodes []*api.Node, image cloud.Image) {
 			klog.Errorf("Error creating node in registry: %v", err)
 			continue
 		}
+		var arch = c.CloudClient.GetArchitecture(newNode.Spec.InstanceType)
+		image, found := images[arch]
+		if !found {
+			klog.Errorf("Error finding image for instance type: %s", newNode.Spec.InstanceType)
+			return
+		}
 		go c.startSingleNode(newNode, image, metadata)
 	}
 }
 
+// TODO create a map[string]string of instances where the key will be the instance
+// type and the value will be its corresponding architecture. This will allow us
+// to match the instance with the proper image for instantiation
+func getImageForInstance(instType string, images []cloud.Image) (cloud.Image, bool) {
+	for _, img := range images {
+		isMacInst := strings.HasPrefix(instType, "mac")
+		if isMacInst && img.Architecture == cloud.Arch_x86_64_mac {
+			return img, true
+		}
+		if !isMacInst && img.Architecture == cloud.Arch_x86_64 {
+			return img, true
+		}
+	}
+	return cloud.Image{}, false
+}
```

Comment:
I think this should go in the instanceselector package, that's the one that knows about instance families and types.